test/xdg_desktop_portal_test.dart

```diff
@@ -82,6 +82,35 @@ class MockCamera {
   String toString() => '$runtimeType($options)';
 }
 
+class MockDevice {
```

Comment:
This represents a request, not a particular device (it could be a request for multiple devices). So probably better named `MockDeviceRequest`